src/Dialect/ONNX/ONNXOpsHelper.cpp

```diff
@@ -87,3 +87,25 @@ bool getIntegerLiteralFromValue(Value value, int64_t &intLit) {
   }
   return false;
 }
+
+//===----------------------------------------------------------------------===//
+// Get a broadcasted type for RankedTensorType and MemRefType.
+//===----------------------------------------------------------------------===//
+Type getBroadcastedRankedType(Type type1, Type type2) {
+  if (type1.isa<RankedTensorType>() && type2.isa<RankedTensorType>())
+    return OpTrait::util::getBroadcastedType(type1, type2);
+  if (type1.isa<MemRefType>() && type2.isa<MemRefType>()) {
+    // Contruct RankedTensorType(s).
+    auto elementType = type1.cast<MemRefType>().getElementType();
```

Comment:
If you don't mind having explicit types here, would be appreciated.

Reply:
Done. Thanks!